File: resources/expected/ros2/isolette_inverted_topics/lax/src/isolette_cpp_pkg/src/user_code/thermostat_monitor_temperature_manage_monitor_mode_mmmt_src.cpp
#include "isolette_cpp_pkg/user_headers/thermostat_monitor_temperature_manage_monitor_mode_mmmt_src.hpp"
#include "isolette_cpp_pkg/base_headers/enum_converter.hpp"

//===========================================================
// This file will not be overwritten when re-running Codegen
//===========================================================

//=================================================
//  I n i t i a l i z e    E n t r y    P o i n t
//=================================================
void thermostat_monitor_temperature_manage_monitor_mode_mmmt::initialize()
{
    PRINT_INFO("Initialize Entry Point invoked");

    // Initialize the node
    // Initialize the node's incoming data port values here
    isolette_cpp_pkg_interfaces::msg::TempWstatusimpl current_tempWstatus = isolette_cpp_pkg_interfaces::msg::TempWstatusimpl();
    init_current_tempWstatus(current_tempWstatus);

    isolette_cpp_pkg_interfaces::msg::FailureFlagimpl interface_failure = isolette_cpp_pkg_interfaces::msg::FailureFlagimpl();
    init_interface_failure(interface_failure);

    isolette_cpp_pkg_interfaces::msg::FailureFlagimpl internal_failure = isolette_cpp_pkg_interfaces::msg::FailureFlagimpl();
    init_internal_failure(internal_failure);

}

//=================================================
//  C o m p u t e    E n t r y    P o i n t
//=================================================
void thermostat_monitor_temperature_manage_monitor_mode_mmmt::timeTriggered()
{
    // Handle communication

    // example receiving messages on data ports
    isolette_cpp_pkg_interfaces::msg::TempWstatusimpl::SharedPtr current_tempWstatus = get_current_tempWstatus();
    PRINT_INFO("Received current_tempWstatus: %s", MESSAGE_TO_STRING(current_tempWstatus));
    isolette_cpp_pkg_interfaces::msg::FailureFlagimpl::SharedPtr interface_failure = get_interface_failure();
    PRINT_INFO("Received interface_failure: %s", MESSAGE_TO_STRING(interface_failure));
    isolette_cpp_pkg_interfaces::msg::FailureFlagimpl::SharedPtr internal_failure = get_internal_failure();
    PRINT_INFO("Received internal_failure: %s", MESSAGE_TO_STRING(internal_failure));

    // Example publishing messages
    isolette_cpp_pkg_interfaces::msg::MonitorMode monitor_mode = isolette_cpp_pkg_interfaces::msg::MonitorMode();
    put_monitor_mode(monitor_mode);
}

File: resources/expected/ros2/python-isolette/strict/src/isolette_py_pkg/isolette_py_pkg/user_code/thermostat_monitor_temperature_manage_monitor_mode_mmmt_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from rosidl_runtime_py.convert import message_to_yaml
from isolette_py_pkg.base_code.thermostat_monitor_temperature_manage_monitor_mode_mmmt_base_src import thermostat_monitor_temperature_manage_monitor_mode_mmmt_base
from isolette_py_pkg_interfaces.msg import TempWstatusimpl
from isolette_py_pkg_interfaces.msg import FailureFlagimpl
from isolette_py_pkg_interfaces.msg import MonitorMode
from isolette_py_pkg.base_code.enum_converter import *

#===========================================================
# This file will not be overwritten when re-running Codegen
#===========================================================
class thermostat_monitor_temperature_manage_monitor_mode_mmmt(thermostat_monitor_temperature_manage_monitor_mode_mmmt_base):
    def __init__(self):
        super().__init__()
        # invoke initialize entry point
        self.initialize()

        self.get_logger().info("thermostat_monitor_temperature_manage_monitor_mode_mmmt infrastructure set up")

#=================================================
#  I n i t i a l i z e    E n t r y    P o i n t
#=================================================
    def initialize(self):
        self.get_logger().info("Initialize Entry Point invoked")

        # Initialize the node

        # Initialize the node's incoming data port values here

        # Initialize the node's incoming data port values here
        current_tempWstatus = TempWstatusimpl()
        self.init_current_tempWstatus(current_tempWstatus)

        interface_failure = FailureFlagimpl()
        self.init_interface_failure(interface_failure)

        internal_failure = FailureFlagimpl()
        self.init_internal_failure(internal_failure)


#=================================================
#  C o m p u t e    E n t r y    P o i n t
#=================================================
    def message_to_string(self, msg):
        yaml_str = message_to_yaml(msg)
        return yaml_str

    def timeTriggered(self):
        # Handle communication

        # Example receiving messages on data ports
        current_tempWstatus = self.get_current_tempWstatus()
        self.get_logger().info(f"Received current_tempWstatus: {self.message_to_string(current_tempWstatus)}")

        interface_failure = self.get_interface_failure()
        self.get_logger().info(f"Received interface_failure: {self.message_to_string(interface_failure)}")

        internal_failure = self.get_internal_failure()
        self.get_logger().info(f"Received internal_failure: {self.message_to_string(internal_failure)}")


        # Example publishing messages
        monitor_mode = MonitorMode()
        self.put_monitor_mode(monitor_mode)


#=================================================
#  Include any additional declarations here
#=================================================
# Additions within these tags will be preserved when re-running Codegen

# Additions within these tags will be preserved when re-running Codegen
